### PR TITLE
Sheffield | 26-Jan-ITP | Daniel Aderibigbe | Sprint 3 | Implement and rewrite test done

### PR DESCRIPTION
**Pull request by Dan2Clouted**
## Learners, PR Template

Self checklist

- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] My changes meet the requirements of the task
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

## Changelist
Completed all the exercisces required.

### CODE REVIEW

**Review by cjyuan — COMMENTED** [review]
Function implementations look good.

This exercise has a second part **"2 Rewrite tests with Jest"** ==> three more files to be updated.

**Comment by Dan2Clouted** [comment]
The have been updated now. Thanks in Advance

File: Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/1-get-angle-type.test.js [hunk 1]
```diff
@@ -14,7 +14,33 @@ test(`should return "Acute angle" when (0 < angle < 90)`, () => {
...
+  expect(getAngleType(181)).toEqual("Reflex angle");
+  expect(getAngleType(270)).toEqual("Reflex angle");
+  expect(getAngleType(359)).toEqual("Reflex angle");
+});
+
 // Case 6: Invalid angles
+test(`should return "Invalid angle" when angle < 0 or angle >= 360`, () => {
+  expect(getAngleType(-1)).toEqual("Invalid angle");
+  expect(getAngleType(360)).toEqual("Invalid angle");
+});
```
Comment (cjyuan) [inline Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/1-get-angle-type.test.js:46]: It's good that you tested one of the boundary cases. You could consider testing the other one.

File: Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/2-is-proper-fraction.test.js [hunk 2]
```diff
@@ -1,10 +1,37 @@
...
+test(`should return false when the numerator is equal to or greater than the denominator`, () => {
+  expect(isProperFraction(2, 2)).toEqual(false);
+  expect(isProperFraction(5, 4)).toEqual(false);
+});
+
+// Proper fractions with negative values
+test(`should return true when the absolute value of the numerator is smaller than the absolute value of the denominator`, () => {
+  expect(isProperFraction(-1, 2)).toEqual(true);
+  expect(isProperFraction(1, -2)).toEqual(true);
+});
```
Comment (cjyuan) [inline Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/2-is-proper-fraction.test.js:24]: This category probably cover the positive case, and you can use notations like `| ... |` or `abs(...)` in the description to make it more concise.

**Review by cjyuan — COMMENTED** [review]
Changes look good. Just a few suggestions.

I will mark this as Complete first.

**Comment by illicitonion** [comment]
Closing PR because the January ITP run has finished. Feel free to re-open if you're still working on it.
